vendor: upgrade go-geom to pickup more GEOMETRYCOLLECTION EMPTY fixes

otan opened the pull request:
This allows us to further differentiate GEOMETRYCOLLECTION EMPTY from
GEOMETRYCOLLECTION Z/ZM/M EMPTY.

Release justification: low risk change to new functionality

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61240)
<!-- Reviewable:end -->

otan committed:
vendor: upgrade go-geom to pickup more GEOMETRYCOLLECTION EMPTY fixes

This allows us to further differentiate GEOMETRYCOLLECTION EMPTY from
GEOMETRYCOLLECTION Z/ZM/M EMPTY.

Release justification: low risk change to new functionality

Release note: None

andyyang890 reviewed (approved):
Thanks for fixing this!

otan commented:
bors r=andyyang890
